be/src/env/env.h

```diff
@@ -41,6 +44,12 @@ class Env {
     Env() = default;
     virtual ~Env() = default;
 
+    static void Register(std::string_view pattern, FactoryFunc func);
+
+    static StatusOr<std::unique_ptr<Env>> CreateUniqueFromString(std::string_view uri);
```

Comment:
CreateFromString is a better name.

Reply:
It's highly possible to return a std::shared_ptr<Env> in the future.